### PR TITLE
Added CI for single toolchain channel workspaces

### PR DESCRIPTION
**Pull request by UebelAndre**
This provides regression testing for cases where users have only registered one toolchain in their workspace (meaning they only have a toolchain of a single [channel](https://rust-lang.github.io/rustup/concepts/channels.html) type).

### CODE REVIEW

File: test/process_wrapper/rustc_quit_on_rmeta.rs [hunk 1]
```diff
@@ -68,8 +68,7 @@ mod test {
         ]);
         assert!(
             !out_content.contains("should not be in output"),
-            "output should not contain 'should not be in output' but did: {}",
-            out_content
+            "output should not contain 'should not be in output' but did",
```
Comment (UebelAndre) [inline test/process_wrapper/rustc_quit_on_rmeta.rs:71]: I believe this change is related to https://github.com/rust-lang/rust-clippy/pull/10055

Unfortunately `#[allow(clippy::uninlined_format_args]` cannot be used here without requiring we raise our min supported Rust version above 1.59.0

File: test/cargo_build_script/tools_exec.rs [hunk 2]
```diff
@@ -1,9 +1,9 @@
+
 #[test]
 pub fn test_tool_exec() {
     let tool_path = env!("TOOL_PATH");
     assert!(
         tool_path.contains("-exec-"),
-        "tool_path did not contain '-exec-': {}",
-        tool_path
+        "tool_path did not contain '-exec-'",
```
Comment (UebelAndre) [inline test/cargo_build_script/tools_exec.rs:7]: I believe this change is related to https://github.com/rust-lang/rust-clippy/pull/10055

Unfortunately `#[allow(clippy::uninlined_format_args]` cannot be used here without requiring we raise our min supported Rust version above 1.59.0

File: test/process_wrapper/rustc_quit_on_rmeta.rs (hunk 1 above)
Comment (illicitonion) [reply test/process_wrapper/rustc_quit_on_rmeta.rs:71]: Can `rustversion::attr` be used here? https://crates.io/crates/rustversion

(I don't know the answer - I know `rustversion` can be used for conditional attrs on structs, and can't be used for conditional enabling of features - I'm not sure how `allow`s fit in :))

Comment (UebelAndre) [reply test/process_wrapper/rustc_quit_on_rmeta.rs:71]: I don't think it'd be appropriate to add a new external dependency just for this. It seems it'll be fixed in a future nightly release so I think it's an acceptable trade off for now. If I can remember it I'll add them back in the future

**Review by illicitonion — APPROVED** [review]

**Pull request merged**